tests/unit/sys/test_expressionparser.cxx

```diff
@@ -213,20 +213,25 @@ TEST_F(ExpressionParserTest, BadNumbers) {
 }
 
 TEST_F(ExpressionParserTest, BadFunctions) {
-  EXPECT_THROW(parser.parseString("sin(--)"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x, )"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x = 1)"), ParseException);
+  parser.addGenerator("increment", std::make_shared<IncrementGenerator>());
+
+  EXPECT_THROW(parser.parseString("increment(--)"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x, )"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x = 1)"), ParseException);
 }
 
 TEST_F(ExpressionParserTest, BadExpressions) {
+  parser.addGenerator("increment", std::make_shared<IncrementGenerator>());
+
   EXPECT_THROW(parser.parseString("x = 1"), ParseException);
-  EXPECT_THROW(parser.parseString("sin(x"), ParseException);
-  EXPECT_THROW(parser.parseString("sin[x"), ParseException);
+  EXPECT_THROW(parser.parseString("increment(x"), ParseException);
+  EXPECT_THROW(parser.parseString("increment"), ParseException);
   EXPECT_THROW(parser.parseString("2]"), ParseException);
   EXPECT_THROW(parser.parseString("4+"), ParseException);
   EXPECT_THROW(parser.parseString("+4"), ParseException);
   EXPECT_THROW(parser.parseString("\n"), ParseException);
   EXPECT_THROW(parser.parseString("(3"), ParseException);
+  EXPECT_THROW(parser.parseString("2-3[4"), ParseException);
```

Comment:
Maybe this replaces the above `[` that was removed?

Reply:
Not quite, but it does hit a different error [here](https://github.com/boutproject/BOUT-dev/blob/next/src/sys/expressionparser.cxx#L271)